client/src/pages/PaymentPortal.js

```diff
@@ -28,20 +32,31 @@ class PaymentPortal extends Component {
   componentDidMount(){
     //TODO: Figure out how it is being passed later : this.props.houseHoldID
     //TODO: Add logic for retrieving householdID
-    //TODO: Replact 5c8680ffad46ec4f26e7b46f with householdID from redux
-    let urlToFetch = API_STUDENT_PROD+ '5c8680ffad46ec4f26e7b46f';
+
+
+    const householdId = this.props.auth.household.householdId;
+    const idFromJwt = jwt_decode(localStorage.getItem('jwtToken'));
+    
+    this.setState({
+      householdId: householdId
+    })
+
+    if (idFromJwt.householdId == this.state.householdId) {
```

Comment:
`this.setState` is called asynchronously, so by this point, `this.state` has not updated. You need to use `householdId` rather than `this.state.householdId` from here so it's current.